cpp/log/cert.cc

```diff
@@ -883,7 +914,8 @@ Cert::Status Cert::IsValidNameConstrainedIntermediateCa() const {
 
   // If it's not a CA cert or there is no name constraint extension then we
   // don't need to apply the rules any further
-  if (HasBasicConstraintCATrue() != Cert::TRUE
+  StatusOr<bool> has_ca_constraint_status = HasBasicConstraintCATrue();
```

Comment:
Noted. BTW I'm used to a type system where bool is not eligible for implicit integer conversion, which steadies hands a good deal more!